### PR TITLE
Verify existence of docker image strings at submit time

### PR DESCRIPTION
**Pull request by helgridly**

### CODE REVIEW

**Comment by geoffjentry** [comment]
squash

File: herc/docker.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        r = self._token_request(user, repo)
+
+        if r.status_code == 404:
+            raise HTTPError(400, "Docker image {user}/{repo} not found".format(user=user, repo=repo))
+        else:
+            r.raise_for_status()  # this will force something weird like a 500 to bubble up to the user
+
+        # Docker doesn't seem to mind old (recent) tokens, so it's no biggie if we race here and a different thread
+        # immediately overwrites this. (Dict assigns are atomic.)
+        DockerTools.tokens[user+'/'+repo] = r.headers['x-docker-token']
```
Comment (geoffjentry) [inline herc/docker.py]: What does PEP8 say about spaces in [user+'/'+repo] (that's an actual question and not a passive aggressive comment)

**Comment by helgridly** [comment]
I will squash the final commit into the previous one but IMO multiple commits is fine in principle.

File: herc/docker.py (hunk 1 above)
Comment (helgridly) [inline herc/docker.py]: It says it's [fine](https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements)

**Comment by geoffjentry** [comment]
So IMO a 'commit' at the PR level should be a full unit of work. It's possible that a branch would have multiple but intermediate steps, smaller stuff, etc would be part of a larger whole. So in this case all 4 commits are part of "Verify existence of docker image strings at submit time"

**Pull request closed without merging**